### PR TITLE
Overline refactor

### PR DESCRIPTION
**Pull request by brentmiller1973**
## Description

This refactors to the overline block to use the overline component (instead of the overline component in shared styles), advanced styles and it's own intl tokens (instead of intl token in shared styles).

The corresponding feature pack PR is here https://github.com/WPMedia/arc-themes-feature-pack/pull/260

## Jira Ticket

- [[TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-735)

## Acceptance Criteria

Use the overline component.
Use its own internationalization token(s)…instead of the ones in shared styles
Add styling for the block in the news theme file

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout themes-735-overline-block`
2. You may need to run `npm i` to pull in `arc-themes-release-version-2.0.1` instead of the 2.0.0 release.
3. Run `npm run storybook` and validate the overline block.

## Dependencies or Side Effects

**Need reviewers to ensure I set up the internationalization tokens correctly.  I believe I did, just want another pair of eyes on it**

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ x] Confirmed all the test steps a reviewer will follow above are working.
- [x ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [ x] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/overline-block/features/overline/default.jsx [hunk 1]
```diff
@@ -1,11 +1,84 @@
...
-import { Overline as OverlineOutput } from "@wpmedia/shared-styles";
+const BLOCK_CLASS_NAME = "b-overline";
 
-const Overline = () => <OverlineOutput />;
+const Overline = (props) => {
+	const { globalContent, arcSite } = useFusionContext();
+	const { editableContent } = useEditableContent();
+	const phrases = getTranslatedPhrases(getProperties(arcSite).locale || "en");
+
+	const { customText, customUrl, editable, story, className = "" } = props;
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.jsx]: This block should have no props

File: blocks/overline-block/features/overline/default.jsx [hunk 2]
```diff
@@ -1,11 +1,84 @@
...
+
+	const sourceContent = story || globalContent || {};
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.jsx]: This block relies on `globalContent` only

File: blocks/overline-block/features/overline/default.jsx [hunk 3]
```diff
@@ -1,11 +1,84 @@
...
+		}
+	}
+
+	const classNames = [BLOCK_CLASS_NAME];
+	const itemProps = {
+		...edit,
+		href: url ? formatURL(url) : null,
+	};
+
+	if (className) {
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.jsx]: This is not needed as we do not allow class names to be passed into this block

File: blocks/overline-block/features/overline/default.jsx [hunk 4]
```diff
@@ -1,11 +1,84 @@
...
+		<OverlineOutput {...itemProps} suppressContentEditableWarning>
+			{text}
+		</OverlineOutput>
+	) : null;
+};
 
 Overline.label = "Overline – Arc Block";
 
-Overline.icon = "arc-overline";
+Overline.propTypes = {
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.jsx]: No need for `propTypes` as this block is not to be consumed

File: blocks/overline-block/features/overline/default.test.jsx [hunk 5]
```diff
@@ -1,10 +1,427 @@
+// import React from "react";
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.test.jsx]: Can we removed commented out code

File: .storybook/themes/news.scss [hunk 6]
```diff
@@ -502,6 +502,18 @@
 					font-size: var(--global-font-size-13),
 					font-weight: var(--global-font-weight-7),
 				),
+				overline: (
```
Comment (matthewroach) [inline .storybook/themes/news.scss]: We don't need any of these values as they are all the same as the component values set above

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/overline-block/features/overline/default.test.jsx (hunk 5 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.test.jsx]: removed.

File: blocks/overline-block/features/overline/default.jsx (hunk 4 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.jsx]: Prop types and the logic associated with them have been removed

File: blocks/overline-block/features/overline/default.jsx (hunk 3 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.jsx]: classNames removed

File: blocks/overline-block/features/overline/default.jsx (hunk 2 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.jsx]: I believe I narrowed it down to global content logic only

File: blocks/overline-block/features/overline/default.jsx (hunk 1 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.jsx]: No more props

File: .storybook/themes/news.scss (hunk 6 above)
Comment (brentmiller1973) [reply .storybook/themes/news.scss]: Kept the declaration for clarity but removed the values.  Did the same for the feature pack PR

File: blocks/overline-block/index.story.mdx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+import Overline from "./features/overline/default";
+
+<Meta title="Blocks/Overline" component={Overline} />
+
+# Byline
+
+The `Overline` component reads from globalContent and returns an overline as a link or span.
+
+## Usage
```
Comment (matthewroach) [inline blocks/overline-block/index.story.mdx]: We shouldn't document usage examples here as blocks are only to be consumed via PageBuilder and not other blocks/code

File: blocks/overline-block/features/overline/default.test.jsx [hunk 8]
```diff
@@ -1,10 +1,188 @@
...
+	useEditableContent: jest.fn(() => ({
+		editableContent: jest.fn(() => {}),
+	})),
+}));
+
+describe("overline feature for default output type", () => {
+	describe("when website_section content from globalContent is present", () => {
+		it("should render an a", () => {
+			const { container } = render(<Overline />);
+			expect(container.querySelectorAll("a")).toHaveLength(1);
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.test.jsx]: This is testing the implementation of the underlying component - if the component changes the HTML element this test will fail.

File: blocks/overline-block/features/overline/default.test.jsx [hunk 9]
```diff
@@ -1,10 +1,188 @@
...
+		});
+
+		it("should dangerously set the inner HTML to the website_section content", () => {
+			const { container } = render(<Overline />);
+			expect(container.querySelector("a").textContent).toContain("News");
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.test.jsx]: The test should not be querying the DOM to get the text. RTL allows you to query by text `screen.getByText('News')`

File: blocks/overline-block/features/overline/default.test.jsx [hunk 10]
```diff
@@ -1,10 +1,188 @@
...
+								_id: "/mock/",
+								name: "Mock",
+							},
+						},
+					},
+				},
+			};
+			useFusionContext.mockImplementation(() => mockStory);
+			const { container } = render(<Overline />);
+			expect(container.querySelector("span").textContent).toContain(
```
Comment (matthewroach) [inline blocks/overline-block/features/overline/default.test.jsx]: Same note as above - we should not be querying the DOM to get text - we should use RTL's queries to ensure we are not writing brittle tests

**Review by matthewroach — CHANGES_REQUESTED** [review]
We should update the tests to not be some brittle and use RTL queries

File: blocks/overline-block/package.json [hunk 11]
```diff
@@ -5,11 +5,11 @@
   "author": "Beltran Caliz <beltran.caliz@washpost.com>",
```
Comment (matthewroach) [inline blocks/overline-block/package.json]:
```suggestion
  "author": "Arc XP - Themes",
```

File: blocks/overline-block/index.story.mdx (hunk 7 above)
Comment (brentmiller1973) [reply blocks/overline-block/index.story.mdx]: removed usage examples

File: blocks/overline-block/features/overline/default.test.jsx (hunk 8 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.test.jsx]: Tests have been updated

File: blocks/overline-block/features/overline/default.test.jsx (hunk 9 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.test.jsx]: Tests have been updated

File: blocks/overline-block/features/overline/default.test.jsx (hunk 10 above)
Comment (brentmiller1973) [reply blocks/overline-block/features/overline/default.test.jsx]: Updated

**Review by matthewroach — APPROVED** [review]

**Pull request merged**